modules/homa-lite/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+	#[pallet::storage]
+	#[pallet::getter(fn batch_total_issuance_info)]
+	pub type BatchTotalIssuanceInfo<T: Config> =
+		StorageMap<_, Blake2_128Concat, EraIndex, TotalIssuanceInfo, OptionQuery>;
+
+	/// The batch that is currency active
+	/// CurrentBatch: value: batch: EraIndex
+	#[pallet::storage]
+	#[pallet::getter(fn current_batch)]
+	pub type CurrentBatch<T: Config> = StorageValue<_, EraIndex, ValueQuery>;
+
+	/// The account in which the staking currency goes into to be transferred to the Relay chain.
+	/// RelaychainStashAccount: value: stash_account: AccountId
+	#[pallet::storage]
+	#[pallet::getter(fn relaychain_stash_account)]
+	pub type RelaychainStashAccount<T: Config> = StorageValue<_, T::AccountId, OptionQuery>;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Request to mint some Liquid currency, by locking up the given amount of Staking
+		/// currency. The exchange does not happen immediately, but on when the batch is processed
+		/// The user then needs to manually claim the Liquid currency once it is ready.
+		///
+		/// Parameters:
+		/// - `amount`: The amount of Staking currency to be exchanged.
+		//#[pallet::weight(< T as Config >::WeightInfo::request_mint())]
+		#[pallet::weight(0)]
+		#[transactional]
+		pub fn request_mint(origin: OriginFor<T>, amount: Balance) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			let current_batch = Self::current_batch();
+			let staking_currency_id = T::StakingCurrencyId::get();
+
+			let stash_account = &Self::relaychain_stash_account().ok_or(Error::<T>::RelaychainStashAccountNotSet)?;
+
+			// TODO: Cross-chain transfer to the Relaychain via XCM
+			T::Currency::transfer(staking_currency_id, &who, &stash_account.clone(), amount)?;
+
+			let current = PendingAmount::<T>::get(current_batch, &who);
+			// Due to the math on Total Issuance, it is literally impossible to overflow here.
+			let new_value = current + amount;
+			PendingAmount::<T>::insert(current_batch, &who, new_value);
+
+			Self::deposit_event(Event::<T>::MintRequested(current_batch, who, amount));
+			Ok(().into())
+		}
+
+		/// Process a batch.
+		/// It is then that we can issue Liquid currencies.
+		///
+		/// Parameters:
+		/// - `staking_total_issuance`:
+		//#[pallet::weight(< T as Config >::WeightInfo::issue())]
+		#[pallet::weight(0)]
+		#[transactional]
+		pub fn issue(origin: OriginFor<T>, staking_total_issuance: Balance) -> DispatchResultWithPostInfo {
+			T::IssuerOrigin::ensure_origin(origin)?;
+			let current_batch = Self::current_batch();
+
+			ensure!(
+				staking_total_issuance > 0,
+				Error::<T>::InvalidStakedCurrencyTotalIssuance
+			);
+
+			let total_liquid_issuance = T::Currency::total_issuance(T::LiquidCurrencyId::get());
+			let total_for_batch = TotalIssuanceInfo {
+				staking_total: staking_total_issuance,
+				liquid_total: total_liquid_issuance,
+			};
```

Comment:
Nitpicks but I think the variables naming consistency could be improved. Rename `staking_total_issuance` to `staking_total`, `total_liquid_issuance` to `liquid_total`, and your `TotalIssuanceInfo` construction would be

```rust
TotalIssuanceInfo { staking_total, liquid_total }
```